src/upload.ts

```diff
@@ -116,6 +172,132 @@ export async function uploadFileRequest(
   return response;
 }
 
+/**
+ * Uploads a large file to Skynet using tus.
+ *
+ * @param this - SkynetClient
+ * @param file - The file to upload.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointLargeUpload="/skynet/tus"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The returned skylink.
+ * @throws - Will throw if the request is successful but the upload response does not contain a complete response.
+ */
+export async function uploadLargeFile(
+  this: SkynetClient,
+  file: File,
+  customOptions?: CustomUploadOptions
+): Promise<UploadRequestResponse> {
+  // Validation is done in `uploadLargeFileRequest`.
+
+  const response = await this.uploadLargeFileRequest(file, customOptions);
+
+  // Sanity check.
+  validateLargeUploadResponse(response);
+
+  // Get the skylink.
+  const metadata = response.headers["upload-metadata"];
+  // Convert the string metadata header into a map.
+  const metadataMap: Map<string, string> = new Map(metadata.split(",").map((pair: string) => pair.split(" ")));
+  let skylink = metadataMap.get("Skylink");
+  // Validate that metadata contains Skylink.
+  if (!skylink) {
+    throw new Error("Response header 'upload-metadata' missing 'Skylink' field");
+  }
+  skylink = Buffer.from(skylink, "base64").toString("utf-8");
+
+  // Get the remaining fields.
+  const siaSkylink = SiaSkylink.fromString(skylink);
+  const merkleroot = toHexString(siaSkylink.merkleRoot);
+  const bitfield = siaSkylink.bitfield;
+
+  // Format the skylink.
+  skylink = formatSkylink(skylink);
+
+  return { skylink, merkleroot, bitfield };
+}
+
+/**
+ * Makes a request to upload a file to Skynet.
+ *
+ * @param this - SkynetClient
+ * @param file - The file to upload.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointLargeUpload="/skynet/tus"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The upload response.
+ */
+export async function uploadLargeFileRequest(
+  this: SkynetClient,
+  file: File,
+  customOptions?: CustomUploadOptions
+): Promise<AxiosResponse> {
+  // TODO: Accept Buffer in Node?
+  // validateFile("file", file, "parameter");
+  validateOptionalObject("customOptions", customOptions, "parameter", defaultUploadOptions);
+
+  const opts = { ...defaultUploadOptions, ...this.customOptions, ...customOptions };
+
+  // TODO: Add back upload options once they are implemented in skyd.
+  const url = await buildRequestUrl(this, opts.endpointLargeUpload);
+  const headers = buildRequestHeaders(undefined, opts.customUserAgent, opts.customCookie);
+
+  // file = ensureFileObjectConsistency(file);
+  // let filename = file.name;
+  let filename = "asdf";
```

Comment:
Forgotten debug?

Reply:
I had a feeling I'd forget about that and leave it in. 😀